pkg/scheduler/scheduler.go

```diff
@@ -281,10 +286,33 @@ func New(client clientset.Interface,
 	sched.StopEverything = stopEverything
 	sched.client = client
 
-	addAllEventHandlers(sched, informerFactory)
+	// Build dynamic client and dynamic informer factory
+	var dynInformerFactory dynamicinformer.DynamicSharedInformerFactory
+	// options.kubeConfig can be nil in tests.
+	if options.kubeConfig != nil {
+		dynClient := dynamic.NewForConfigOrDie(options.kubeConfig)
+		dynInformerFactory = dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynClient, 0, v1.NamespaceAll, nil)
```

Comment:
In terms of default scheduler, there is no plugin registering CR events, so IMO instantiating a `dynInformerFactory` won't do any concrete work.

Reply:
yes, just like coreInformerFactory, by default the embedded informers are empty.

Reply:
hmm, so someone needs to initiate an informer then, so that makes it even more necessary to pass this factory via framework handler to the plugins that will use it, right?

Reply:
Yes, that's your point at https://github.com/kubernetes/kubernetes/pull/101394#discussion_r620264539. For now, plugin developers instantiate **typed** informer and register event handlers accordingly.

Reply:
It is related, but might be different. If you remember, an informer is initiated only if it is explicitly accessed before starting the the informer. Is this the case here (I don't have experience with dynamic informer, so asking just to make sure)?

Reply:
> If you remember, an informer is initiated only if it is explicitly accessed before starting the the informer.

This is true. However, the framework doesn't really care about the cache for CRs - the out-of-tree plugins are responsible to build their own cache, if needed. The hook setup here is fairly simple: just to guarantee that upon some CR events, partial unschedulable Pods can be moved properly.

(BTW: I rethink the necessity of offering the dynInformerFactory in framework handle. See my other comment https://github.com/kubernetes/kubernetes/pull/101394#discussion_r620773413)